### PR TITLE
Speaker embeddings demo improvements

### PR DESCRIPTION
**Pull request by michalkulakowski**
### 🛠 Summary

JIRA/Issue if applicable.
Describe the changes.

### 🧪 Checklist

- [ ] Unit tests added.
- [ ] The documentation updated.
- [ ] Change follows security best practices.
``

### CODE REVIEW

File: demos/audio/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+torch
```
Comment (dkalinowski) [inline demos/audio/requirements.txt]: use cpu only versions so it doesnt download cuda, check out other demos of ours, same issue appears in openwebui demo

File: demos/audio/create_speaker_embedding.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import torchaudio
+from speechbrain.inference.speaker import EncoderClassifier
+import sys
+
+file = sys.argv[1]
+signal, fs = torchaudio.load(file)
+if signal.shape[0] > 1:
+    signal = torch.mean(signal, dim=0, keepdim=True)
+expected_sample_rate = 16000
+if(fs != expected_sample_rate):
```
Comment (dkalinowski) [inline demos/audio/create_speaker_embedding.py]:
```suggestion
if fs != expected_sample_rate:
```

File: demos/audio/README.md [hunk 3]
```diff
@@ -101,14 +78,48 @@ ovms --rest_port 8000 --source_model microsoft/speecht5_tts --model_repository_p
...
+
+print("Generation finished")
+```
+:::
+
+:::{dropdown} **Unary call with curl**
+
+
+```bash
+curl http://localhost:8000/v3/audio/speech -H "Content-Type: application/json" -d "{\"model\": \"microsoft/speecht5_tts\", \"voice\":\"voice1\", \"input\": \"The quick brown fox jumped over the lazy dog\"}" -o speech.wav
```
Comment (dkalinowski) [inline demos/audio/README.md:119]: did you test all of that with sphinx?

File: demos/audio/README.md [hunk 4]
```diff
@@ -16,6 +16,16 @@ Check supported [Speech Recognition Models](https://openvinotoolkit.github.io/op
 **Client**: curl or Python for using OpenAI client package
 
 ## Speech generation
+### Prepare speaker embeddings
+When generating speech you can use default speaker voice (only with curl) or you can prepaere your own speaker embedding file. Here you can see how to do it with downloaded file from online repository, but you can try with your own speech recorded as well:
+```bash
+pip install -r requirements.txt
```
Comment (porlows1) [inline demos/audio/README.md]: pip install -r pip3 install -r https://raw.githubusercontent.com/openvinotoolkit/model_server/refs/heads/main/demos/audio/requirements.txt

**Review by dkalinowski — APPROVED** [review]

File: demos/audio/create_speaker_embedding.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+# Copyright (C) 2026 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import torchaudio
+from speechbrain.inference.speaker import EncoderClassifier
+import sys
+
+file = sys.argv[1]
```
Comment (rasapala) [inline demos/audio/create_speaker_embedding.py:14]: check arg size

File: demos/audio/create_speaker_embedding.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+signal, fs = torchaudio.load(file)
+if signal.shape[0] > 1:
```
Comment (rasapala) [inline demos/audio/create_speaker_embedding.py:16]: check shape size

File: demos/audio/create_speaker_embedding.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+if fs != expected_sample_rate:
+    resampler = torchaudio.transforms.Resample(orig_freq=fs, new_freq=expected_sample_rate)
+    signal = resampler(signal)
+
+classifier = EncoderClassifier.from_hparams(source="speechbrain/spkrec-xvect-voxceleb")
+embedding = classifier.encode_batch(signal)
+embedding = torch.nn.functional.normalize(embedding, dim=2)
+embedding = embedding.squeeze().cpu().numpy().astype("float32")
+
+output_file = sys.argv[2]
```
Comment (rasapala) [inline demos/audio/create_speaker_embedding.py:35]: check arg size

File: demos/audio/README.md [hunk 8]
```diff
@@ -16,6 +16,17 @@ Check supported [Speech Recognition Models](https://openvinotoolkit.github.io/op
 **Client**: curl or Python for using OpenAI client package
 
 ## Speech generation
+### Prepare speaker embeddings
+When generating speech you can use default speaker voice or you can prepare your own speaker embedding file. Here you can see how to do it with downloaded file from online repository, but you can try with your own speech recorded as well:
```
Comment (rasapala) [inline demos/audio/README.md]:
```suggestion
When generating speech you can use default speaker voice or you can prepare your own speaker embedding file. Here you can see how to do it with downloaded file from online repository, but you can try with your own speech recording as well:
```

File: demos/audio/README.md [hunk 9]
```diff
@@ -16,6 +16,17 @@ Check supported [Speech Recognition Models](https://openvinotoolkit.github.io/op
...
+```bash
+pip install -r pip3 install -r https://raw.githubusercontent.com/openvinotoolkit/model_server/refs/heads/main/demos/audio/requirements.txt
+mkdir -p audio_samples
+curl --output audio_samples/audio.wav "https://www.voiptroubleshooter.com/open_speech/american/OSR_us_000_0032_8k.wav"
+mkdir -p models
```
Comment (rasapala) [inline demos/audio/README.md:25]:
```suggestion
```

File: demos/audio/requirements.txt [hunk 10]
```diff
@@ -0,0 +1,5 @@
+--extra-index-url "https://download.pytorch.org/whl/cpu"
+torch==2.9.1+cpu
+torchaudio==2.9.1+cpu
+speechbrain
```
Comment (rasapala) [inline demos/audio/requirements.txt]: Add specific versions used in demo to make sure we do not break when dependencies update components.
speechbrain=
openai=

File: demos/common/export_models/export_model.py [hunk 11]
```diff
@@ -91,6 +91,9 @@ def add_common_arguments(parser):
 add_common_arguments(parser_text2speech)
 parser_text2speech.add_argument('--num_streams', default=0, type=int, help='The number of parallel execution streams to use for the models in the pipeline.', dest='num_streams')
 parser_text2speech.add_argument('--vocoder', type=str, help='The vocoder model to use for text2speech. For example microsoft/speecht5_hifigan', dest='vocoder')
+parser_text2speech.add_argument('--speaker_name', type=str, help='Name of the speaker', dest='speaker_name')
```
Comment (rasapala) [inline demos/common/export_models/export_model.py:94]: When making changes to export_model we should also incorporate them into ovms --pull and ovms parameters. If this applies here please file a jira to plan changes in ovms cpp.

**Review by rasapala — DISMISSED** [review]

File: demos/common/export_models/export_model.py (hunk 11 above)
Comment (michalkulakowski) [reply demos/common/export_models/export_model.py:94]: CVS-181526

**Review by mzegla — APPROVED** [review]
Please check copilot suggestions
